test/e2e/prometheus_test.go

```diff
@@ -5300,6 +5300,70 @@ func testPrometheusStatusScale(t *testing.T) {
 	}
 }
 
+func testPrometheusServiceName(t *testing.T) {
+	t.Parallel()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	name := "test-servicename"
+
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-service", name),
+			Namespace: ns,
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeLoadBalancer,
+			Ports: []v1.ServicePort{
+				{
+					Name: "web",
+					Port: 9090,
+				},
+			},
+			Selector: map[string]string{
+				"prometheus":                   name,
+				"app.kubernetes.io/name":       "prometheus",
+				"app.kubernetes.io/instance":   name,
+				"app.kubernetes.io/managed-by": "prometheus-operator",
+			},
+		},
+	}
+
+	if _, err := framework.CreateOrUpdateService(context.Background(), ns, svc); err != nil {
+		t.Fatal(fmt.Errorf("failed to create app service: %w", err))
+	}
+
+	pm := framework.MakeBasicPodMonitor(name)
```

Comment:
do we really need to create a pod monitor and check the targets?

Reply:
I think I rushed it at the time, maybe we could use `framework.PrometheusSVCGetRequest()`.